pkg/cloudprovider/cloudprovider.go

```diff
@@ -321,12 +322,21 @@ func (c *CloudProvider) instanceToMachine(ctx context.Context, ec2instance *ec2.
 	labels[v1alpha1.LabelInstanceAMIID] = aws.StringValue(ec2instance.ImageId)
 	labels[v1.LabelTopologyZone] = aws.StringValue(ec2instance.Placement.AvailabilityZone)
 	labels[v1alpha5.LabelCapacityType] = instance.GetCapacityType(ec2instance)
+	labels[v1.LabelOSStable] = instance.GetOS(ec2instance)
```

Comment:
Double-checking, but I'm assuming that this needs to be done because we no longer can know the OS based on instance type requirements?

Reply:
Not only double-checking. :)

Karpenter returns "Linux" so it breaks the drift cycle in some cases.